File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02581/p02581_num9_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> cards(3*N);
    for(int i=0; i<3*N; i++) {
        cin >> cards[i];
    }
    
    sort(cards.begin(), cards.end());
    
    int points = 0;
    
    for(int i=0; i<N-1; i++) {
        if(cards[3*i] == cards[3*i+2]) {
            points++;
        }
    }
    
    int count = 1;
    for(int i=1; i<3; i++) {
        if(cards[i] == cards[i-1]) {
            count++;
        }
    }
    
    if(count == 3) {
        points++;
    }
    
    cout << points << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02581/p02581_num9_parsed.py
def calculate_points(N, cards):
    points = 0
    i = 0
    while i < N-1:
        if cards[i] == cards[i+1] == cards[i+2] == cards[i+3] == cards[i+4]:
            points += 1
            i += 5
        else:
            i += 1
    
    if cards[N-1] == cards[N] == cards[N+1]:
        points += 1
    
    return points

N = int(input())
cards = list(map(int, input().split()))

print(calculate_points(N, cards))
